### PR TITLE
Initial Infineon platform support

### PR DESCRIPTION
**Pull request by ItoddchouFX**

### CODE REVIEW

**Comment by ItoddchouFX** [comment]
@jwhui Please help review this PR.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+CMakeCache.txt
+CMakeFiles
+cmake_install.cmake
+configure
+config.log
+config.status
+doc/Doxyfile
+doc/html
+etc/cmake/openthread-config-generic.h
+include/openthread-config.h
```
Comment (jwhui) [inline .gitignore:42]: Many of these files do not pertain to this repo. Can we clean this up?

Example: https://github.com/openthread/ot-nrf528xx/blob/main/.gitignore

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib)
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
+
+option(VERBOSE "Enable verbose message" OFF)
+
+# Add platform specific crypto files to openthread stack SDK
+file(
+        GLOB
```
Comment (jwhui) [inline CMakeLists.txt:51]: style: there seems to be a lot of excessive indent here. Can we make this closer to the style seen in other OpenThread files?

Example: https://github.com/openthread/openthread/blob/main/CMakeLists.txt

File: README.md [hunk 3]
```diff
@@ -5,9 +5,92 @@
...
+Download and install the [GNU toolchain for ARM Cortex-M][gnu-toolchain].
+
+[gnu-toolchain]: https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/gnu-rm
+
+In a Bash terminal, follow these instructions to install the GNU toolchain and other dependencies.
+
+```bash
+$ cd <path-to-ot-ifx>
+$ git submodule init
+$ git submodule update
```
Comment (jwhui) [inline README.md:40]: nit: reduce to single command
```suggestion
$ git submodule update --init
```

File: script/build [hunk 4]
```diff
@@ -27,43 +27,192 @@
...
 
-main "$@"
+###################################################################################################
+#
+# Program Start
+#
+###################################################################################################
+
+#set -euxo pipefail
+set -euo pipefail
```
Comment (jwhui) [inline script/build:122]: Can we keep the `x`?

File: script/flash [hunk 5]
```diff
@@ -0,0 +1,137 @@
+#!/bin/bash
+#
+#  Copyright (c) 2020, The OpenThread Authors.
```
Comment (jwhui) [inline script/flash:3]:
```suggestion
#  Copyright (c) 2021, The OpenThread Authors.
```

File: src/cyw30739/alarm.c [hunk 6]
```diff
@@ -0,0 +1,418 @@
+/*
+ * Copyright 2016-2021, Cypress Semiconductor Corporation (an Infineon company) or
```
Comment (jwhui) [inline src/cyw30739/alarm.c:2]: All code not in third-party should retain the OpenThread license header.

File: src/cyw30739/packet_statistics.h [hunk 7]
```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2016-2021, Cypress Semiconductor Corporation (an Infineon company) or
```
Comment (jwhui) [inline src/cyw30739/packet_statistics.h:2]: All files not in third-party should retain the OpenThread license header.

**Review by jwhui — COMMENTED** [review]
Thanks for this initial PR! I completed an initial review and submitted some high-level comments.

**Comment by jwhui** [comment]
@ItoddchouFX , FYI - the CI checks seem to be failing https://github.com/ItoddchouFX/ot-ifx/actions/runs/1716723725.

**Comment by ItoddchouFX** [comment]
> 

Fixed.

**Comment by ItoddchouFX** [comment]
Here is the CI checks. https://github.com/ItoddchouFX/ot-ifx/runs/4876073726?check_suite_focus=true

Below error is a expected error as mentioned in https://github.com/openthread/openthread/issues/7102#issuecomment-1007157927
[✖] https://github.com/openthread/ot-ifx/workflows/Build/badge.svg?branch=main&event=push → Status: 404 

@jwhui Can you review it for merge?

**Review by jwhui — APPROVED** [review]

**Pull request merged**